community/kernel/src/main/java/org/neo4j/kernel/impl/api/cursor/TxAbstractPropertyCursor.java

```diff
@@ -94,30 +92,123 @@ public Object value()
     @Override
     public boolean booleanValue()
     {
-        return (Boolean) value();
-    }
+        Object value = value();
+        if ( value instanceof Boolean )
+        {
+            return (Boolean) value;
+        }
+        else
+        {
+            throw new IllegalStateException( "Value is not a boolean" );
+        }    }
+
+    @Override
+    public byte byteValue()
+    {
+        Object value = value();
+        if ( value instanceof Byte )
```

Comment:
Can we use a generic method that takes a class instead of all this very similar methods?
```
<T> T getValue( Class<T> clazz )
    {
        Object value = value();
        if ( clazz.isInstance( value ) )
        {
            return clazz.cast( value );
        }
        throw new IllegalStateException( "Value is not a " + clazz.getSimpleName() );
    }
```